node/lib/util/git_util.js

```diff
@@ -360,7 +360,13 @@ exports.push = co.wrap(function *(repo, remote, source, target, force, quiet) {
     const execString = `\
 git -C '${repo.path()}' push ${forceStr} ${remote} ${source}:${target}`;
     try {
-        const result = yield ChildProcess.exec(execString, {env : environ});
+        let result = yield ChildProcess.exec(execString, {env : environ});
+        if (result.error &&
```

Comment:
do we need to set a max number of retries here? 
Retry without sleeping or limit make me nervous